pkg/cmd/set/set_env.go

```diff
@@ -0,0 +1,722 @@
+	}
+	if len(o.Filenames) == 0 && len(o.resources) < 1 {
+		return fmt.Errorf("one or more resources must be specified as <resource> <name> or <resource>/<name>")
+	}
+	if o.List && len(o.output) > 0 {
+		return fmt.Errorf("--list and --output may not be specified together")
+	}
+	if len(o.Keys) > 0 && len(o.From) == 0 {
+		return fmt.Errorf("when specifying --keys, a configmap or secret must be provided with --from")
+	}
+	return nil
+}
+
+// RunEnv contains all the necessary functionality for the OpenShift cli env command
+func (o *EnvOptions) RunEnv(f cmdutil.Factory) error {
+	env, remove, err := envutil.ParseEnv(append(o.EnvParams, o.envArgs...), o.In)
+
+	if err != nil {
+		return err
+	}
+
+	if len(o.From) != 0 {
+		b := o.builder().
+			WithScheme(scheme.Scheme, scheme.Scheme.PrioritizedVersionsAllGroups()...).
+			LocalParam(o.Local).
+			ContinueOnError().
+			NamespaceParam(o.namespace).DefaultNamespace().
+			FilenameParam(o.enforceNamespace, &o.FilenameOptions).
+			Flatten()
+
+		if !o.Local {
+			b = b.
+				LabelSelectorParam(o.Selector).
+				ResourceTypeOrNameArgs(o.All, o.From).
+				Latest()
+		}
+
+		infos, err := b.Do().Infos()
+		if err != nil {
+			return err
+		}
+
+		for _, info := range infos {
+			switch from := info.Object.(type) {
+			case *v1.Secret:
+				for key := range from.Data {
+					if contains(key, o.Keys) {
+						envVar := v1.EnvVar{
+							Name: keyToEnvName(key),
+							ValueFrom: &v1.EnvVarSource{
+								SecretKeyRef: &v1.SecretKeySelector{
+									LocalObjectReference: v1.LocalObjectReference{
+										Name: from.Name,
+									},
+									Key: key,
+								},
+							},
+						}
+						env = append(env, envVar)
+					}
+				}
+			case *v1.ConfigMap:
+				for key := range from.Data {
+					if contains(key, o.Keys) {
+						envVar := v1.EnvVar{
+							Name: keyToEnvName(key),
+							ValueFrom: &v1.EnvVarSource{
+								ConfigMapKeyRef: &v1.ConfigMapKeySelector{
+									LocalObjectReference: v1.LocalObjectReference{
+										Name: from.Name,
+									},
+									Key: key,
+								},
+							},
+						}
+						env = append(env, envVar)
+					}
+				}
+			default:
+				return fmt.Errorf("unsupported resource specified in --from")
+			}
+		}
+	}
+
+	if len(o.Prefix) != 0 {
+		for i := range env {
+			env[i].Name = fmt.Sprintf("%s%s", o.Prefix, env[i].Name)
+		}
+	}
+
+	b := o.builder().
+		WithScheme(scheme.Scheme, scheme.Scheme.PrioritizedVersionsAllGroups()...).
+		LocalParam(o.Local).
+		ContinueOnError().
+		NamespaceParam(o.namespace).DefaultNamespace().
+		FilenameParam(o.enforceNamespace, &o.FilenameOptions).
+		Flatten()
+
+	if !o.Local {
+		b.LabelSelectorParam(o.Selector).
+			ResourceTypeOrNameArgs(o.All, o.resources...).
+			Latest()
+	}
+
+	infos, err := b.Do().Infos()
+	if err != nil {
+		return err
+	}
+
+	if len(infos) == 0{
+		return nil
+	}
+	resourceType := strings.Split(infos[0].ObjectName(), "/")[0]
```

Comment:
Only support format like `kubectl-kruise set env cloneset/xxxx  env1=value1` ? 

How to handle it if  I want to use `kubectl-kruise set env clone xxxx env1=value1` ?

Reply:
kubectl-kruise set env clone xxxx env1=value1 gets the expected result, same with the above one.